common/compositor/factory.cpp

```diff
@@ -17,46 +17,106 @@
 #include "factory.h"
 #include "platformdefines.h"
 
-#ifdef USE_GL
 #include "glrenderer.h"
 #include "glsurface.h"
 #include "nativeglresource.h"
-#elif USE_VK
```

Comment:
Right, I'm suggesting that we add Vulkan as a hard build requirement. So Mesa does need to be built with Vulkan enabled, and the Vulkan loader library needs to be installed. Is there some scenario where that would be an impossible configuration? Again, this requirement doesn't mean that we have to change the build configuration based on the target platform.